src/FindTranslocations.cpp

```diff
@@ -8,15 +9,167 @@
 #include <vector>
 #include <map>
 #include <string>
+#include <queue>
 
 #include <sstream>
 #include <iostream>
 #include <fstream>
 
+
+
 #include "data_structures/Translocation.h"
 #include  <boost/program_options.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include "api/BamWriter.h"
+
 namespace po = boost::program_options;
 
+//returns a bam file contaning the reads located in a given region of the genome
+void TranslocationBAM(string BamFileName,string outputFileHeader,string inputFileName,map<string,unsigned int> contig2position){
+	//the number of regions involved in each event
+	int n=2;
+	int j=0;
+	string tmpstr;
+	string line;
+	string eventbam;
+	string eventfolder;
+
+	//creates a folder for the outputdata
+	boost::filesystem::path dir(outputFileHeader.c_str());
+	if(!boost::filesystem::create_directory(dir)) {
+		std::cout << "error creating output folder" << "\n";
+		return;
+	}
+	eventfolder=outputFileHeader+"/events";
+	boost::filesystem::path eventdir( eventfolder.c_str());
+	if(!boost::filesystem::create_directory(eventdir)) {
+		std::cout << "error creating output folder" << "\n";
+		return;
+	}
+
+	//opens one writer for writing one file containing all the output sequences, and one writer to create files containing each separate region
+	BamWriter regionwriter;
+	BamWriter writer;
+	BamReader bamFile;
+	eventbam=outputFileHeader+"/all_events_roi.bam";
+	//open the bam file
+	bamFile.Open(BamFileName);
+	BamAlignment currentRead;
+
+	// retrieve 'metadata' from BAM files, these are required by BamWriter
+	const SamHeader header = bamFile.GetHeader();
+	const RefVector references = bamFile.GetReferenceData();
+
+	// attempt to open our BamWriter{
+	cout << eventbam << endl;
+	if ( !writer.Open(eventbam.c_str(), header, references) ) {
+		cout << "Could not open output BAM file" << endl;
+		return;
+	}
+
+
+
+
+
+	//opens the input file and reads each line
+	cout << "initiates extraction, please wait" << endl;
+	ifstream inputFile( inputFileName.c_str() );
+	if (inputFile){
+		while (getline( inputFile, line )){
+			//skips the input header
+			if (j > 0){
```

Comment:
Suggestion: declare the variable close to their first use, in this way you can be sure that they are properly initialise. Here there are 50 lines of code in between j declaration and the first use of j